### PR TITLE
#1416: Display relationships between resources in the details info panel

### PR DESCRIPTION
**Pull request by dsuren1**
### Description
This PR adds the possibility to show linked resources tab when resource has linked resources

### Issue
- #1416 

### Screenshot
image

### CODE REVIEW

File: geonode_mapstore_client/client/js/components/DetailsPanel/DetailsInfo.jsx [hunk 1]
```diff
@@ -124,6 +125,12 @@ function DetailsInfoFields({ fields, formatHref }) {
     </div>);
 }
 
+const tabTypes = {
+    'attributes': null,
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/components/DetailsPanel/DetailsInfo.jsx:129]: let's remove this one

```suggestion
```

File: geonode_mapstore_client/client/js/components/DetailsPanel/DetailsInfo.jsx [hunk 2]
```diff
@@ -134,24 +141,31 @@ const parseTabItems = (items) => {
         return true;
     });
 };
+const isDefaultTabType = (type) => type === 'tab';
 
 function DetailsInfo({
     tabs = [],
-    formatHref
+    formatHref,
+    types
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/components/DetailsPanel/DetailsInfo.jsx:149]:
```suggestion
    resourceTypesInfo
```

File: geonode_mapstore_client/client/js/components/DetailsPanel/DetailsRelatedResources.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+DetailsRelatedResources.propTypes = {
+    fields: PropTypes.array,
+    types: PropTypes.object
+};
+
+DetailsRelatedResources.defaultProps = {
+    fields: [],
+    types: {}
```
Comment (allyoucanmap) [inline geonode_mapstore_client/client/js/components/DetailsPanel/DetailsRelatedResources.jsx:43]:
```suggestion
    resourceTypesInfo: {}
```

**Review by allyoucanmap — CHANGES_REQUESTED** [review]
@dsuren1 The implementation looks fine, just some minor changes.
I'm not sure if I cover all the replacement for resourceTypesInfo so please double check and please let me know if it makes sense for you this change

**Review by giohappy — CHANGES_REQUESTED** [review]
@dsuren1 sorry about that but we changed our mind again. Let's rename "related resources" to "linked resources", for the variables names, the component, etc. 
It's more natural and it will make it easier to map the client to the backend properties.

File: geonode_mapstore_client/client/js/api/geonode/v2/index.js [hunk 4]
```diff
@@ -306,6 +306,11 @@ export const getResourceByPk = (pk) => {
         .then(({ data }) => data.resource);
 };
 
+export const getLinkedResourcesByPk = (pk) => {
+    return axios.get(parseDevHostname(`${endpoints[RESOURCES]}/${pk}/linked_resources`))
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply geonode_mapstore_client/client/js/api/geonode/v2/index.js:310]: @giohappy 
The api throws following error when setting params `{page: 1, page_size: 99999}`. Could you please look into it
image

**Review by allyoucanmap — APPROVED** [review]

File: geonode_mapstore_client/static/mapstore/gn-translations/data.en-US.json [hunk 5]
```diff
@@ -321,7 +321,8 @@
             "viewFullMetadata": "View full metadata",
             "abstract": "Abstract",
             "readMore": "Read more",
-            "readLess": "Read less"
+            "readLess": "Read less",
+            "relatedResources": "Linked Resources"
```
Comment (allyoucanmap) [inline geonode_mapstore_client/static/mapstore/gn-translations/data.en-US.json:325]: This should be changed to linkedResources

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: geonode_mapstore_client/client/js/api/geonode/v2/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (giohappy) [reply geonode_mapstore_client/client/js/api/geonode/v2/index.js]: @dsuren1 the fix has been merged and it's been deployed to development demo

**Review by allyoucanmap — APPROVED** [review]

**Review by giohappy — APPROVED** [review]

**Comment by giohappy** [comment]
@gannebamm we're already on it. @mattiagiupponi is implementing the reverse relationships to show up inside linked resources.

**Comment by mattiagiupponi** [comment]
@gannebamm @giohappy 
The issue is available here: https://github.com/GeoNode/geonode/issues/10993
And it's related PR is here waiting for review: https://github.com/GeoNode/geonode/pull/10994
Any hint is welcome

**Comment by giohappy** [comment]
@gannebamm deployed to demo if you want to test it

image

**Comment by gannebamm** [comment]
That was fast 😄 
Now it is working as expected (https://development.demo.geonode.org/catalogue/#/dataset/1003):
grafik

awesome!

**Pull request merged**